### DISCUSSION TITLE
DBeaver prompts for driver download even if the driver is already present (but only first time)

### DISCUSSION THREAD

**Discussion by steromano87** [dbeaver/dbeaver#27] [Q&A] [answered]
**Prerequisite**
- You have a working connection that requires the download of its driver from Maven repository.
- The previous connection is closed before executing this test (i.e. the connection was closed when DBeaver was shut down the last time)

Suppose you have a working connection on a MySQL DB.
When you open DBeaver and click on _New Connection..._ button in the DB Navigator panel and select (in our example) a MySQL connection, DBeaver prompts you for the driver download, even if the driver is already present in the system.

If you dismiss this dialog, close the _New Connection_ dialog and open it again, no driver download dialog is shown.

This error happens also if you right-click an existing connection and select the _Modify connection_ option from the menu. Additionally, if you accept and download (but the download really happens? It doesn't seem so...) and save the connection modifications, next time DBeaver is started if you modify the connection again the driver download dialog is prompted again.

You can find a screenhot of the dialog attached to this issue.

![dbeaver_driver_dialog_error](https://cloud.githubusercontent.com/assets/8759452/10794278/7b0bbf0c-7d94-11e5-8b81-bf87ccd5af60.png)

**Comment by serge-rider** [comment]
This happens in 3.5.2 if you've upgraded from 3.5.1. Maven references had changed in some drivers and now there are two different artifacts (good one and bad one). Actually it is a bug (I had to make some migration procedure but this issue didn't appear in my dev environment).

Anyhow, here is a workaround:
- go to the driver editor (in driver manager or just click on Edit Driver button in download dialog)
- you will see two different Maven artifacts in the library list
- click on "Reset to defaults" - only good one should remain
- click "Ok"
- restart DBeaver (just in case)
  This dummy download dialog won't appear again (at least it shouldn't).

**Comment by steromano87** [comment]
Thank you for the workaround, Serge! I confirm that doing so the problem disappears :-)

**Comment by michael-newsrx** [comment]
Doesn't work for me. (LINUX)

**Comment by serge-rider** [comment]
Is this issue still remains in 3.5.4?
Could you provide a bit more details?

**Comment by swapna2109** [comment]
Hi,
It is not working for me in windows 7.
Could you please help?

**Comment by serge-rider** [comment]
Please clarify - what exactly isn't working?

**Comment by swapna2109** [comment]
I followed below steps:- 
1. Clicked on Driver Manager. Then clicked on Edit Driver button
2. Clicked on "Reset to defaults" and then on OK button
3. Restarted DBeaver. The dummy download dialog is appearing again

**Comment by serge-rider** [comment]
This is very old issue and I can't reproduce it.
Most likely your problem is a result of version upgrade (and some config mess).
Try to delete file `%HOMEPATH%\.dbeaver\.metadata\.plugins\org.jkiss.dbeaver.core\drivers.xml` and restart DBeaver. Download dialog should appear just once after that.

**Comment by swapna2109** [comment]
Hi,
Thanks... now its working fine.

**Comment by mrmonterrosa** [comment]
To install the mysql driver manually we go to https://mvnrepository.com/artifact/mysql/mysql-connector-java and choose the version we want in my case I will take version 5.1.47 this is the direct link to download http://central.maven.org/maven2/mysql/mysql-connector-java/5.1.47/mysql-connector-java-5.1.47.jar I also leave the following image so that you can see where you can download it .

![1](https://user-images.githubusercontent.com/17989537/44415445-0c6bab80-a536-11e8-865c-d61b6528cccf.png)

Then we save the driver in directory easy to find.

We do not go DBeaver and we realize how it feels
1)
![1](https://user-images.githubusercontent.com/17989537/44414818-5f446380-a534-11e8-8f66-fe9b5faebe97.png)

2)
![1](https://user-images.githubusercontent.com/17989537/44415673-a29fd180-a536-11e8-90fc-cc7c25495cca.png)

3) The default driver is removed
![1](https://user-images.githubusercontent.com/17989537/44415776-dc70d800-a536-11e8-8424-27624e0b5363.png)

4)  the new downloaded driver is added and everything else is left the same, we give it in **OK**
![1](https://user-images.githubusercontent.com/17989537/44416002-6c168680-a537-11e8-8290-53f4111d482f.png)

5) and finally we fill in the connection data and we make the connection successful
![1](https://user-images.githubusercontent.com/17989537/44416377-55bcfa80-a538-11e8-8e99-9f474915984b.png)

6) If all the data is correct, the following message should appear, then OK > OK
![1](https://user-images.githubusercontent.com/17989537/44416451-8866f300-a538-11e8-9869-7fd1e77ed180.png)

I hope it is of your help.

**Comment by asm-web** [comment]
> To install the mysql driver manually we go to https://mvnrepository.com/artifact/mysql/mysql-connector-java and choose the version we want in my case I will take version 5.1.47 this is the direct link to download http://central.maven.org/maven2/mysql/mysql-connector-java/5.1.47/mysql-connector-java-5.1.47.jar I also leave the following image so that you can see where you can download it .
> 
> ![1](https://user-images.githubusercontent.com/17989537/44415445-0c6bab80-a536-11e8-865c-d61b6528cccf.png)
> 
> Then we save the driver in directory easy to find.
> 
> We do not go DBeaver and we realize how it feels
> 1)
> ![1](https://user-images.githubusercontent.com/17989537/44414818-5f446380-a534-11e8-8f66-fe9b5faebe97.png)
> 
> 
> ![1](https://user-images.githubusercontent.com/17989537/44415673-a29fd180-a536-11e8-90fc-cc7c25495cca.png)
> 
> 1. The default driver is removed
>    ![1](https://user-images.githubusercontent.com/17989537/44415776-dc70d800-a536-11e8-8424-27624e0b5363.png)
> 2. the new downloaded driver is added and everything else is left the same, we give it in **OK**
>    ![1](https://user-images.githubusercontent.com/17989537/44416002-6c168680-a537-11e8-8290-53f4111d482f.png)
> 3. and finally we fill in the connection data and we make the connection successful
>    ![1](https://user-images.githubusercontent.com/17989537/44416377-55bcfa80-a538-11e8-8e99-9f474915984b.png)
> 4. If all the data is correct, the following message should appear, then OK > OK
>    ![1](https://user-images.githubusercontent.com/17989537/44416451-8866f300-a538-11e8-9869-7fd1e77ed180.png)
> 
> I hope it is of your help.

Thank you so much! this post helped me !

**Comment by serge-rider** [comment, marked as answer]
@asm-web You don't need to do this. You can choose any driver version you need right in dbeaver interface.
Go to Download/Update screen, then click on driver version column and choose version you need.
